R/upkeep.R

```diff
@@ -16,30 +38,97 @@ use_tidy_upkeep_issue <- function(year = NULL) {
     }
   }
 
-  checklist <- upkeep_checklist(year)
+  checklist <- if (tidy) tidy_upkeep_checklist(year) else upkeep_checklist()
 
-  maybe_year <- if (is.null(year)) "" else glue(" ({year})")
+  title_year <- year %||% format(Sys.Date(), "%Y")
 
   gh <- gh_tr(tr)
   issue <- gh(
     "POST /repos/{owner}/{repo}/issues",
-    title = glue("Upkeep for {project_name()}{maybe_year}"),
+    title = glue("Upkeep for {project_name()} ({title_year})"),
     body = paste0(checklist, "\n", collapse = "")
   )
   Sys.sleep(1)
   view_url(issue$html_url)
 }
 
-upkeep_checklist <- function(year = NULL,
-                             posit_pkg = is_posit_pkg(),
-                             posit_person_ok = is_posit_person_canonical()) {
-  year <- year %||% 2000
+upkeep_checklist <- function() {
+  bullets <- c(
+    todo("`usethis::use_readme_rmd()`", !file_exists(proj_path("README.Rmd"))),
+    todo("`usethis::use_roxygen_md()`", !is_true(uses_roxygen_md())),
+    todo("`usethis::use_github_links()`", !has_github_links()),
+    todo("`usethis::use_pkgdown_github_pages()`", !uses_pkgdown()),
+    todo(
+      "
+      `usethis::use_package_doc()`
+      Consider letting usethis manage your `@importFrom` directives here. \\
+      `usethis::use_import_from()` is handy for this.",
+      !has_package_doc()
+    ),
+    todo(
+      "
+      `usethis::use_testthat()`. \\
+      Learn more about testing at https://r-pkgs.org/tests.html",
+      !uses_testthat()
+    ),
+    todo(
+      "
+      `usethis::use_testthat(3)` and upgrade to 3e, \\
+      [testthat 3e vignette](https://testthat.r-lib.org/articles/third-edition.html)",
+      uses_old_testthat_edition(current = 3)
+    ),
+    todo("
+      Align the names of `R/` files and `test/` files for workflow happiness. \\
+      The docs for `usethis::use_r()` include a helpful script. \\
+      `usethis::rename_files()` may be be useful."),
+    todo("`usethis::use_github_action('check-standard')`", !uses_github_actions()),
+    todo(
+      "Consider changing default branch from `master` to `main`",
+      git_default_branch() == "master"
+    ),
+    todo("`usethis::use_code_of_conduct()`", !has_coc()),
+    todo(
+      "Modernize citation files; see `usethis::use_citation()`",
+      has_citation_file()
+    ),
+    todo(
+      "Remove any description of check environments from `cran-comments.md`",
+      file_exists(proj_path("cran-comments.md"))
+    ),
+    todo(
+      "
+      Use lifecycle instead of artisanal deprecation messages, as described \\
+      in [Communicate lifecycle changes in your functions](https://lifecycle.r-lib.org/articles/communicate.html)",
+      !proj_desc()$has_dep("lifecycle")
+    ),
```

Comment:
This feels a bit more tidyverse-y than the other bullets.

Reply:
Do you mean the suggestion to use lifecycle itself, or the way it's worded?

Reply:
I suspect he means that this is arguably too opinionated for the public checklist.

Do we really mean to condition on the _absence_ of lifecycle?

You could imagine conditioning on the _presence_ of lifecycle, i.e. target only folks who have clearly opted in to the lifecycle lifestyle. And making it more about checking if anything needs to be advanced.

Reply:
Ah yes, I like the idea of conditioning on the presence of lifecycle. There is then overlap with `use_release_issue()`, though is there a case to be made that upkeep is a better place for it than as you're preparing for a release?  Happy to remove it from here though if just residing in the release checklist is the way to go.

Reply:
I think remove it. Opt-in into the lifecycle lifestyle is not something for every package (IMO most packages don't need that level of deprecation handling)

Reply:
Sounds good to me.